### ISSUE TITLE
Code review requests

### ISSUE THREAD

**Issue by vantoan19** [synd1k4t/Iterex#13]
Please raise a code review / testing request here whenever you've done something and want other people to see and test

**Comment by synd1k4t** [comment]
I have taken a look at the commit, the animations work fine, though the collision is a bit off, probably because of the sprites. The code looks clean enough, I took a look through the various files to piece together how everything works, the solutions seem alright, I'll still have to study it a bit more to see how to integrate to it.

**Comment by vantoan11** [comment]
Yes, the black box for collision, named as CollisionBox, the white one named TextureBox. Every sprites has both properties. For the CollisionBox, I filtered all coloured pixels of a texture and find its boundary, especially for the animation, because it's a long texture with many images in there, so the boundary may vary depend on the image, that's why I find the smallest boundary that cover all boundaries of the animation.
The per pixel collision is just implemented for other reasons, like when we want to implement firing arrows feature, per pixel collision could be a great way to do so compare to the AABB collision.

**Comment by synd1k4t** [comment]
That's about what I guessed, automatically generating the collision boxes based on transparency is smart, it will definitely lead to a better result than just manually setting the box and centering the sprite on it.
As for projectiles, I plan on using AABB to generalize skills, for example have a hitbox use an entity as it's parent when it is doing a melee attack, or a projectile when one is fired, in both cases the hitbox is just there while the parent plays the animation, I'll set up a prototype on saturday to show what I mean. I think pixel collision would be difficult when for example an entity does an attack animation and we only want to check a small part of it, and since such collision will likely need to check every hitbox with every hurtbox every frame, it can end up using a massive amount of calculations if we don't keep it simple.
